scripts/parse-translation.js

```diff
@@ -1,31 +1,113 @@
-var parseKV = require('parse-kv');
-var fs = require('fs');
-var path = require('path');
+const parseKV = require('parse-kv');
+const fs = require('fs');
+const path = require('path');
 
-var basePath = path.join(__dirname, '../');
+const basePath = path.join(__dirname, '../');
+let SPACE_PADDING = '                                                         ';
+var filenameByKey = {};
 
-module.exports = function (shouldParse) {
-  var fileData = readdir(path.join(basePath, 'game/resource/English/'));
+function checkReplaceTokens (allTooltips, key, recursionCheck, source) {
+  key = key.toLowerCase();
+  recursionCheck = recursionCheck || {};
+  if (recursionCheck[key]) {
+    console.error(recursionCheck);
+    throw new Error('Foun d recursive key: ' + key);
+  }
+  recursionCheck[key] = recursionCheck[key] || 0;
+
+  let value = allTooltips[key];
+  let tokens = value.match(/#{[_\- A-Za-z0-9]+}/g); // alphanumeric, _- and space
```

Comment:
@IBurn36360 this is what i settled on...

Reply:
That should work.  Hopefully Valve could go through can make it a standard some day so the regex can be really tight (Not holding my breath...surviving is enjoyable at times).

Reply:
Half the time the files aren't even formatted properly. Their tooltips file had a malformed line around "Era" and "Kyxy" for like 4 months.

Reply:
I would be utterly shocked if they had test cases. If they have them then they don't run them.

Reply:
It likely is just build success (Compiling the executable) and a few AI bot matches completing, if that.  Other than the regex, nothing stands out to me as potentially a problem.  A dry run would be the best way to be sure though.

Reply:
* Stood out, you already addressed the regex